### DISCUSSION TITLE
Loki: Use custom level field name for log volume

### DISCUSSION THREAD

**Discussion by sherifkayad** [grafana/grafana#43274] [Observability]
**What happened**:
The Volume Histogram can't interpret the levels of the logs and classified logs as `unknown` when the label `level` isn't used but another label e.g. `severity` is used.

The logs panel shows the **correct** coloring though.

**What you expected to happen**:
Both the Volume section and the Logs panel show the right coloring (e.g. Green for Info, Red for Error, .. etc.) .. Maybe Grafana needs to enable a list of fields that the user can define via the data source config for the severity / log level?!

**How to reproduce it (as minimally and precisely as possible)**:
- Use a different field for log level e.g. `severity` and send these logs to Loki
- Query the logs making sure you have the `fullRangeLogsVolume` supported

**Anything else we need to know?**:
Just attaching some screenshots ..
![image](https://user-images.githubusercontent.com/8362784/146007163-a230d021-b084-4203-aea4-4dadca557d3a.png)

![image](https://user-images.githubusercontent.com/8362784/146020617-c57a0cd5-a0a4-47c2-867f-9b05875c3e19.png)

**Environment**:
- Grafana version: 8.3.2
- Data source type & version: Loki v2.4.1
- OS Grafana is installed on: Kubernetes using the official Helm Chart
- User OS & Browser: Chrome
- Grafana plugins:
- Others:

**Comment by sherifkayad** [comment]
@ivanahuckova what do you think? I saw you replied to numerous similar issues. Can you help with this one?

**Comment by Elfo404** [comment]
Hi @sherifkayad
The way the severity is extracted is described in https://grafana.com/docs/grafana/latest/explore/logs-integration/#logs-in-explore, currently we do not support extracting levels from other fields, but you can upvote the feature request at https://github.com/grafana/grafana/issues/25585.

closing this as duplicate of #25585

**Comment by sherifkayad** [comment]
@Elfo404 I am not really sure if this issue here is a duplicate for #25585 .. Here it's not about custom log levels, however, about supporting other fields other than `level` for the severity

**Comment by sherifkayad** [comment]
@Elfo404 also the way described in the docs works for me in the log panel .. just not in the volume histogram and that's why I thought this might be a bug ..

**Comment by sherifkayad** [comment]
@Elfo404 Thanks for re-opening. I attached another screenshot to explain the issue I am having.

**Comment by ifrost** [comment]
The new log volume histogram doesn't support custom log level fields yet, but I think it's a good idea to add it to the configuration. We do have such a configuration for Elastic. It should be considered as a feature request so I think #25585 is still valid (though it was created for old log histogram).

**Comment by ifrost** [comment]
I'll move it to Discussions as it should be considered a feature request. #25585 is related, the configuration should be applicable to both log lines and the histogram aggregation.

**Comment by sherifkayad** [comment]
@ifrost how should we proceed here? Should I open a feature request out of this discussion ticket now?

**Reply by ifrost** [reply]
I've already moved it to Discussions where all feature requests are discussed. I'll also bring it to our team planning session to see if/when it could be scheduled.

**Reply by sherifkayad** [reply]
amazing! Thanks a lot

**Comment by sherifkayad** [comment]
@ifrost any update after the discussions with the team?

**Reply by ifrost** [reply]
@grafana/observability-logs-and-traces: AFAIK it wasn't planned for the next quarter. Do you think it makes sense to add it to the backlog?

**Comment by ivanahuckova** [comment]
For additional discussion, please use https://github.com/grafana/grafana/issues/77557
